addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/RequestorJobUnitTest.java

```diff
@@ -368,6 +368,116 @@ void shouldSendMessageWithSetValues() throws IOException {
+
+        // When
+        job.setJobData(jobData);
+        job.verifyParameters(progress);
+        job.runJob(env, progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(false)));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        ArgumentCaptor<HttpMessage> argument = ArgumentCaptor.forClass(HttpMessage.class);
+        verify(httpSender).sendAndReceive(argument.capture());
+        assertThat(
+                argument.getValue().getRequestHeader().getHeader("header1"), is(equalTo("value1")));
+    }
+
+    @Test
+    void shouldSendHeaderWithEmptyStringValue() throws IOException {
+        // Given
+        AutomationProgress progress = new AutomationProgress();
+        AutomationEnvironment env = new AutomationEnvironment(progress);
+        HttpSender httpSender = mock(HttpSender.class);
+        RequestorJob job = new RequestorJob(httpSender);
+
+        String yamlStr =
+                "requests:\n"
+                        + "- url: https://www.example.com\n"
+                        + "  headers:\n"
+                        + "    header1: ''";
+        Yaml yaml = new Yaml();
+        LinkedHashMap<?, ?> jobData = (LinkedHashMap<?, ?>) yaml.load(yamlStr);
+
+        // When
+        job.setJobData(jobData);
+        job.verifyParameters(progress);
+        job.runJob(env, progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(false)));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        ArgumentCaptor<HttpMessage> argument = ArgumentCaptor.forClass(HttpMessage.class);
+        verify(httpSender).sendAndReceive(argument.capture());
+        assertThat(argument.getValue().getRequestHeader().getHeader("header1"), is(equalTo("")));
+    }
+
+    @Test
+    void shouldSendHeaderWithNullValue() throws IOException {
+        // Given
+        AutomationProgress progress = new AutomationProgress();
+        AutomationEnvironment env = new AutomationEnvironment(progress);
+        HttpSender httpSender = mock(HttpSender.class);
+        RequestorJob job = new RequestorJob(httpSender);
+
+        String yamlStr =
+                "requests:\n"
+                        + "- url: https://www.example.com\n"
+                        + "  headers:\n"
+                        + "    header1: ";
+        Yaml yaml = new Yaml();
+        LinkedHashMap<?, ?> jobData = (LinkedHashMap<?, ?>) yaml.load(yamlStr);
+
+        // When
+        job.setJobData(jobData);
+        job.verifyParameters(progress);
+        job.runJob(env, progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(false)));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        ArgumentCaptor<HttpMessage> argument = ArgumentCaptor.forClass(HttpMessage.class);
+        verify(httpSender).sendAndReceive(argument.capture());
+        assertThat(argument.getValue().getRequestHeader().getHeader("header1"), is(equalTo(null)));
```

Comment:
`null` means no header.

Reply:
The test is called `shouldSendHeaderWithNullValue`.

Reply:
The purpose is to send a header with no value.

Reply:
No value is the empty string.

Reply:
I guess it should actually be testing that a header is removed.

Reply:
As per my understanding of the code, ` assertThat(argument.getValue().getRequestHeader().getHeader("header1"), is(equalTo(null)));` this is matching `null` with the value of the `header1`. If you see the code in `getHeader()` it is returning the value of the header, so header is being sent but with null value or no value. The other unit test is `shouldSendHeaderWithEmptyStringValue` which sends the header with empty string as value so both are different test cases and I have ran these two cases from YAML file and they're working as expected.

Reply:
> so header is being sent but with null value or no value

No, the `null` value means the header is not present in the message, if it's not in the message it's not being sent.

Reply:
Ok, so its something to do with the message code, I'll see it. Should I remove this unit test?

Reply:
What do we want here? I can't see the reason of putting null in YAML but definitely you have more experience so I'll go with your call on this.

Reply:
It's already using "null" in the YAML, we now need to decide how the job will interpret it. IMO it could be the same as empty.

Reply:
Yes its same as empty. I have checked it, the header is not sent with null value.

Reply:
Note that there should be a test with null value (can be in `shouldSendHeadersWithSetValues` though).

Reply:
Its the same test I wrote earlier for null value right? `header1:` like this? At the end we just will be checking that header is removed from request because of null value as you mentioned the message's behavior

Reply:
Yup, that sounds right

Reply:
Ok I will update `shouldSendHeadersWithSetValues`